src/meta/processors/partsMan/CreateSpaceProcessor.cpp

```diff
@@ -18,6 +18,12 @@ void CreateSpaceProcessor::process(const cpp2::CreateSpaceReq& req) {
     auto properties = req.get_properties();
     auto spaceRet = getSpaceId(properties.get_space_name());
     if (spaceRet.ok()) {
+        if (req.get_if_not_exists()) {
+            resp_.set_id(to(spaceRet.value(), EntryType::SPACE));
+            resp_.set_code(cpp2::ErrorCode::SUCCEEDED);
+            onFinished();
+            return;
+        }
```

Comment:
I think you only need to handle the `resp.set_code()` logic according to `req.get_if_not_exists()`.  And the rest is common.  For example:

```cpp
if (req.get_if_not_exists()) {
  resp.set_code(SUCCEED);
} else {
  LOG(ERROR) << "";
  resp.set_code(EXISTED);
}
resp.set_id();
onFinished();
return;
```